src/Blocks/utils.jl

```diff
@@ -14,19 +15,40 @@
     ODESystem(Equation[], t, [u...], []; name = name)
 end
 @doc """
-    RealInput(;name, nin, u_start)
+    RealInput(;name, u_start)
 
 Connector with one input signal of type Real.
 
 # Parameters:
-- `nin=1`: Number of inputs
-- `u_start=0`: Initial value for `u`
+- `u_start=0`: Initial value for `u`.
```

Comment:
Why does this have an initial value default?

Reply:
Shouldn't there be a default value for the variable?

```julia
julia> @named x = RealInput(; nin = 1, u_start = 1.0)
Model x with 0 (1) equations
Unknowns (1):
  u(t) [defaults to 1.0]: Inner variable in RealInput x
Parameters (0):
```

Reply:
Won't that cause issues in initialization?

Reply:
Addressed in [4cfc041](https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/283/commits/4cfc041d9f6d242c48e5a5df41758d89ce23eb25)

Reply:
Removing the default is erroring - https://github.com/SciML/ModelingToolkitStandardLibrary.jl/actions/runs/8706828508/job/23880355450?pr=283#step:6:577 even though there is an initial guess 🤔

Reply:
After talking with Aayush, we found guesses are not propagated in a hierarchical system. Hopefully https://github.com/SciML/ModelingToolkit.jl/pull/2654 should fix it

Reply:
One more bug from MTK - `linearize_function` doesn't use guesses - https://github.com/SciML/ModelingToolkit.jl/pull/2656 should fix it

Reply:
Linearization shouldn't necessarily use guesses. It should make sure the algebraic conditions are satisfied and that's it. That's a separate issue though, just update the test to be good and we can merge, the linearization failure is separate from this.

Reply:
Do you mean to mark it as broken/skip? done that in [9bc4b8b](https://github.com/SciML/ModelingToolkitStandardLibrary.jl/pull/283/commits/9bc4b8bceef58149cd00932bba791b54d208ae0b)

Reply:
no, fix it by passing in state values